src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -3284,6 +3289,92 @@ public void testAndWhereMultipleCheckAndEnclosed() throws DatabaseEngineExceptio
+    public void testStringAggDelimiter() throws DatabaseEngineException {
+        test5Columns();
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "TESTE")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "TeStE")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "tesTte")
+                .build());
+
+        final List<Map<String, ResultColumn>> query = engine.query(
+                select(column("COL1"), stringAgg(column("COL5")).delimiter(';').alias("agg"))
+                        .from(table("TEST"))
+                        .groupby(column("COL1"))
+                        .orderby(column("COL1").asc())
+        );
+
+        assertEquals("Resultset must have only 2 results", 2, query.size());
+        assertEquals("COL1 must be 1", 1, query.get(0).get("COL1").toInt().intValue());
+        assertEquals("COL5 must be TESTE;teste", "TESTE;teste", query.get(0).get("agg").toString());
+        assertEquals("COL1 must be 2", 2, query.get(1).get("COL1").toInt().intValue());
+        assertEquals("COL5 must be TeStE;tesTte", "TeStE;tesTte", query.get(1).get("agg").toString());
+    }
+
+    @Test
+    public void testStringAggDistinct() throws DatabaseEngineException {
+        test5Columns();
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "TeStE")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "tesTte")
+                .build());
+
+        try {
+            final List<Map<String, ResultColumn>> query = engine.query(
+                    select(column("COL1"), stringAgg(column("COL5")).distinct().alias("agg"))
+                            .from(table("TEST"))
+                            .groupby(column("COL1"))
+                            .orderby(column("COL1").asc())
+            );
+
+            assertEquals("Resultset must have only 2 results", 2, query.size());
+            assertEquals("COL1 must be 1", 1, query.get(0).get("COL1").toInt().intValue());
+            assertEquals("COL5 must be teste", "teste", query.get(0).get("agg").toString());
+            assertEquals("COL1 must be 2", 2, query.get(1).get("COL1").toInt().intValue());
+            assertEquals("COL5 must be TeStE,tesTte", "TeStE,tesTte", query.get(1).get("agg").toString());
+
+        } catch (OperationNotSupportedRuntimeException e) {
+            // Ignore for SQL Server and ORACLE since they do not support it.
+            if (!this.engine.getDialect().equals(SQLSERVER) && !this.engine.getDialect().equals(ORACLE)) {
+                throw e;
+            }
+        }
+    }
+
```

Comment:
Please add a test of using string aggregation with columns that are not strings.